hi-ml-cpath/src/health_cpath/configs/classification/BaseMIL.py

```diff
@@ -241,14 +253,33 @@ def get_transforms_dict(self, image_key: str) -> Optional[Dict[ModelKey, Union[C
     def get_encoder_params(self) -> EncoderParams:
         return create_from_matching_params(self, EncoderParams)
 
+    def get_num_classes(self) -> int:
+        """Gets the number of classes that the model handles. In training mode, this is the number of classes in the
+        training dataset. In evaluation mode, this is read directly from the `eval_num_classes` attribute.
+
+        This method has a side effect in training mode: It creates the data module if it is not yet set in the
+        `data_module` attribute.
+        """
+        if self.runner_mode == RunnerMode.EVAL_FULL:
+            # We are in evaluation mode, so we don't necessarilty have access to the training data module. The eval
+            # data module may not contain all classes, so we need to hardcode those values.
+            return self.eval_num_classes
+        else:
+            if (not hasattr(self, "data_module")) or self.data_module is None:
+                self.data_module = self.get_data_module()
+            return self.data_module.train_dataset.n_classes
+
     def create_model(self) -> DeepMILModule:
-        self.data_module = self.get_data_module()
+        num_classes = self.get_num_classes()
+        # Class weights are only used to create a loss function, which is irrelevant in evaluation mode.
+        class_weights = None if self.runner_mode == RunnerMode.EVAL_FULL else self.data_module.class_weights
```

Comment:
wondering if we would still need class_weights those in eval mode to compute the test loss? 
Otherwise YESS, it seems like a good solution

Reply:
eval test loss will be different than training mode, but it's probably ok

Reply:
as discussed, we don't want to use loss for eval, will add a comment